### PR TITLE
SOLR-15111 Use JDK8 Base64 instead of own implementation

### PR DESCRIPTION
**Pull request by asalamon74**
Original lucene-solr pull request: https://github.com/apache/lucene-solr/pull/2252 I removed the lucene part of it.

# Description

JDK8 has a builtin Base64 encoder and decoder, there is no need to use own implementation for this.

# Solution

Eliminate own implementation.

# Tests

Unit tests

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java [hunk 1]
```diff
@@ -298,7 +300,7 @@ private static String getFieldFlags( SchemaField f )
 
       BytesRef bytes = field.binaryValue();
       if (bytes != null) {
-        f.add( "binary", Base64.byteArrayToBase64(bytes.bytes, bytes.offset, bytes.length));
+        f.add( "binary", new String(Base64.getEncoder().encode(ByteBuffer.wrap(bytes.bytes, bytes.offset, bytes.length)).array(), StandardCharsets.ISO_8859_1));
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java:303]: Can be simplified to `Base64.getEncoder().encodeToString(ByteBuffer.wrap(bytes.bytes, bytes.offset, bytes.length).array())`

**Review by janhoy — APPROVED** [review]
Looks mostly straight forward. But I think a few places can be simplified by calling `encodeToString()` rather than `new String(<bytes>, <charset>)`, see comment.

File: solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java (hunk 1 above)
Comment (asalamon74) [reply solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java:303]: I'm afraid that's not exactly the same. ByteBuffer's `array()` method returns the full array and ignores the offset and the length.

**Comment by asalamon74** [comment]
@janhoy Added the CHANGES.txt entry. I executed `./gradlew check` as far as I know it executes the tests.

**Pull request merged**